internal/server/application_message.go

```diff
@@ -0,0 +1,116 @@
+package server
+
+import (
+	"errors"
+	"fmt"
+	"net"
+	"reflect"
+	"sync"
+
+	"github.com/negasus/haproxy-spoe-go/message"
+)
+
+func unmarshalMessage(msg *message.Message, out interface{}) error {
```

Comment:
dont take this personally :D This isnt meant badly, but: WTF?
Please no reflection inside the hotpath, especially when you can hardcode all fields

Reply:
removed it ❤️